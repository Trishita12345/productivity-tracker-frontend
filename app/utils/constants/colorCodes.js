export const COLORS = {
  LIGHT_GREY: '#F5F5F5',
  GREY: '#C6C6C6',
  DARK_GREY: '#87797F',
  VIOLET: '#70367C',
  ORANGE: '#FD7E50',
  SKY: '#95D0D5',
  DARK_VIOLET: '#301137',
  GREEN: '#6DB33F',
  YELLOW: '#DCD41D',
  RED: '#E41515',
  WHITE: '#FFFAFF',
  BLUE: '#4E63B2',
};
