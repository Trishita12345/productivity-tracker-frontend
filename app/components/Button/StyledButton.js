import styled from 'styled-components';
import { COLORS } from '../../utils/constants/colorCodes';

const textColor = {
  primary: COLORS.WHITE,
  secondary: COLORS.DARK_GREY,
};
const bgColor = {
  primary: COLORS.DARK_VIOLET,
  secondary: COLORS.LIGHT_GREY,
};
const bgColorActive = {
  primary: COLORS.BLUE,
  secondary: COLORS.GREY,
};

const StyledButton = styled.button`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 13px 19px;
  color: ${({ variant }) => textColor[variant]};

  background: ${({ variant }) => bgColor[variant]};
  border: ${({ variant }) => bgColor[variant]};
  border-radius: 10px;
  filter: drop-shadow(6px 6px 6px rgba(0, 0, 0, 0.25));

  &:active {
    background: ${({ variant }) => bgColorActive[variant]};
    cursor: pointer;
  }

  &:hover {
    cursor: pointer;
  }
`;

export default StyledButton;
