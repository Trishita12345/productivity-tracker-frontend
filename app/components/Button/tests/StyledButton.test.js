import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import StyledButton from '../StyledButton';
import { VARIANTS } from '../../../utils/constants/variants';
import { COLORS } from '../../../utils/constants/colorCodes';
import 'jest-styled-components';

configure({ adapter: new Adapter() });

describe('<StyledButton />', () => {
  const props = { variant: VARIANTS.PRIMARY };

  const ComponentRenderer = myProps => {
    const component = mount(<StyledButton {...myProps} />);
    return component;
  };

  it('should render with primary variants', () => {
    const component = ComponentRenderer(props);
    expect(component).toHaveStyleRule('color', COLORS.WHITE);
    expect(component).toHaveStyleRule('background', COLORS.DARK_VIOLET);
    expect(component).toMatchSnapshot();
  });
  it('should render with secondary variants', () => {
    const component = ComponentRenderer({
      ...props,
      variant: VARIANTS.SECONDARY,
    });
    expect(component).toHaveStyleRule('color', COLORS.DARK_GREY);
    expect(component).toHaveStyleRule('background', COLORS.LIGHT_GREY);
    expect(component).toMatchSnapshot();
  });
});
