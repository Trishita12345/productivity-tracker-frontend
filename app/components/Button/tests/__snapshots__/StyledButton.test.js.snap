// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledButton /> should render with primary variants 1`] = `ReactWrapper {}`;

exports[`<StyledButton /> should render with secondary variants 1`] = `ReactWrapper {}`;
